docs/src/pages/ShorthandProps.mdx

```diff
@@ -0,0 +1,186 @@
+export const meta = {
+  title: 'Shorthand Props',
+  next: { url: '/quick-start', name: 'Quick Start' },
+}
+
+Stardust UI components can have "slots" which accept shothand props. For example, `Button` component has an `icon` slot which value defines the icon that will be rendered.
+
+```jsx
+<Button icon="camera" />
+```
+
+There are several forms of shorthand values that can be provided, but all of them share one common thing - each is eventually evaluated to React Element. Thus, you can think of shorthand values as a recipe to customize rendered React Element.
+
+## Object as value
+
+Each component's slot has associated default element type. For example, by default there is
+`<Icon />` element rendered for `Button`'s `icon` slot. It is possible to customize props of this default element by providing props object as shorthand value:
+
+```jsx
+// 💡 'name' and 'size' will be used as <Icon /> element's props
+<Button icon={{ name: 'camera', size: 'small' }} />
+```
+
+## Primitive as value
+
+There is even shorter way to define default element's props - by using a primitive value. In that case provided shorthand value will be taken as a value for "default prop" of this element.
+
+This could be much easier seen with an example. Here, again, we have a `Button`
+element with its icon being defined with shorthand - where provided `string` value
+will be used as icon's `name`:
+
+```jsx
+<>
+  <Button icon="camera" />
+  // 💡 has identical effect to the previous one
+  <Button icon={{ name: 'camera' }} />
+</>
+```
+
+This works because `name` is the default prop of shorthand's `<Icon />` element. You can also pass React Elements to slots:
+
+```jsx
+<>
+  <Popup content={<em>Awesome Popup!</em>} />
+  // 💡 has identical effect to the previous one
+  <Popup content={{ content: <em>Awesome Popup!</em> }} />
+</>
+```
+
+### Disable slot's rendering
+
+It is also possible to pass falsy values (`false`, `null` or
+`undefined`) to shorthand prop - iin that case there will be nothing rendered for the component's slot.
+
+```jsx
+<Dropdown toggleIndicator={null} />
+```
+
+## Function as value
+
+Providing function as a shorthand value is the most involving but, at the same time, the most
+powerful option for customizing component's slot. The only requirements for this function are:
+
+- it should finish syncronously
+- it should return React Element as a result
+
+Thus, in its simplest form, it could be used the following way:
+
+```jsx
+<Button icon={() => <Icon name="chess rook" />} />
+```
+
+However, if it will be just about that, this function form wouldn't introduce any additional value. So, lets take a look on the scenarios where it is really helpful.
+
```

Comment:
Not sure why we have this section if we say that it doesn't introduce any additional value.